app.rb

```diff
@@ -22,6 +26,55 @@ def get_bank_code(code)
       end
       sublet_code || regular_code
     end
+
+    # Gets details of banks given the bank name,
+    # city and state
+    def filter_banks(state = nil, city = nil, bank = nil, limit = nil, offset = nil)
+
+      filtered_df = $df
+
+      unless state.nil?
+        filtered_df = filtered_df.where(filtered_df["STATE"].eq(state))
```

Comment:
Use `ISO3166` for searching, since that is a much more sane limited set than all the misspellings that you'd see here.

Reply:
Our current design assumes that the frontend sends the correct state values (user selects from a list), hence the use of `ISO3166` might not be required

Reply:
If you look at the dataset, the `ISO-3166` key is standardized and easier to search. If the frontend is anyway sending the correct value (and we're trusting it), let's ask it to send standard values instead. Otherwise, common misspellings and spelling changes in the future will break things.

Reply:
@captn3m0 Changed the search to use ISO3166